web/src/setup.ts

```diff
@@ -5,8 +5,20 @@ import * as loading from "./loading";
 import {page_params} from "./page_params";
 import * as util from "./util";
 
+// Here we're extending the global JQuery namespace with our custom functions
+// safeOuterHeight and safeOuterWidth. This is done to make typescript stop
+// yelling. Functions $.fn.outerHeight and $.fn.outerWidth have two possible
+// overloads. But we only need the argument types for one of the overloads.
+// That overload is the one which only requires `includeMargin` as the argument.
+declare global {
+    interface JQuery {
+        safeOuterHeight(...args: Parameters<typeof $.fn.outerHeight>): number | 0;
+        safeOuterWidth(...args: Parameters<typeof $.fn.outerWidth>): number | 0;
+    }
+}
```

Comment:
As discussed at https://chat.zulip.org/#narrow/stream/6-frontend/topic/typescript.20migration/near/1529684, we should remove `safeOuterHeight` and `safeOuterWidth` from our code base. Would you like to submit a prep PR to do that?